tests/performance/event_generator/src/event_generator/event_generator.py

```diff
@@ -0,0 +1,240 @@
+    def generate_event(self) -> None:
+        """Generate and write a log event based on a predefined template or a simple default format."""
+        os.makedirs(os.path.dirname(self.path), exist_ok=True)
+        try:
+            self.write_log()
+        except OSError as e:
+            logging.error(f"Error writing to log file: {e}")
+            if not self.retry_write():
+                logging.error("Failed to write to log after several attempts.")
+
+        time.sleep(1)  # Sleep to simulate delay between logs
+
+    def write_log(self) -> None:
+        """Write a log entry to the file based on the current date and time."""
+        current_time = datetime.now()
+        log_data = {
+            "date": current_time.strftime("%Y-%m-%d"),
+            "time": current_time.strftime("%H:%M:%S"),
+            "severity": "INFO",
+            "message": "This is a test log message"
+        }
+        log_entry = self.template_format(log_data)
+        with open(self.path, "a") as log_file:
+            log_file.write(log_entry + "\n")
+        logging.info(f"Log event generated at {self.path}")
+
+        # Check file size and manage if necessary
+        if os.path.getsize(self.path) > self.max_file_size:
+            self.rotate_log()
+
+    def template_format(self, data: dict[str, str]) -> str:
+        """Format the log data based on the template provided during initialization.
+
+        Args:
+            data (dict): Dictionary containing log data.
+
+        Returns:
+            str: A formatted string based on the template.
+        """
+        template_str = json.dumps(self.template)
+        for key, value in data.items():
+            template_str = template_str.replace('{' + key + '}', value)
+        return template_str
+
+    def rotate_log(self) -> None:
+        """Rotate the log file when the size limit is exceeded.
+
+        This creates a new log file by appending a sequence number or timestamp to the filename.
+        """
+        import time
+        timestamp = int(time.time())  # Timestamp for uniqueness
+        new_path = f"{self.path}_{timestamp}.old"
+        os.rename(self.path, new_path)
+        self.path = self.path.replace(
+            ".log", f"_{timestamp}.log")  # Reset to new log file
+        open(self.path, 'w').close()  # Create a new log file
+        logging.info(f"Log file exceeded size limit, rotated to {new_path}")
+
+    def retry_write(self) -> bool:
+        """Attempt to write the log file up to a maximum number of retries.
+
+        Returns:
+            bool: True if writing succeeded, False otherwise.
+        """
+        retry_count = 0
+        max_retries = 3
+        while retry_count < max_retries:
+            try:
+                self.write_log()
+                return True
+            except OSError:
+                retry_count += 1
+                time.sleep(1)  # wait a bit before retrying
+        return False
+
+
+class SyscheckEventGenerator(EventGenerator):
```

Comment:
I would make all methods private except stop and start

Reply:
Fixed here: https://github.com/wazuh/wazuh-qa/pull/5697/commits/70cf83bd07d20a1157bb1cef60c551d24b14f51e